src/prop/prop_engine.cpp

```diff
@@ -478,11 +478,20 @@ Result PropEngine::checkSat() {
   }
 
   Trace("prop") << "PropEngine::checkSat() => " << result << std::endl;
-  if (result == SAT_VALUE_TRUE && d_theoryProxy->isIncomplete())
+  if (result == SAT_VALUE_TRUE)
   {
-    outputIncompleteReason(UnknownExplanation::INCOMPLETE,
-                           d_theoryProxy->getIncompleteId());
-    return Result(Result::UNKNOWN, UnknownExplanation::INCOMPLETE);
+    if (d_theoryProxy->isIncomplete())
+    {
+      outputIncompleteReason(UnknownExplanation::INCOMPLETE,
+                             d_theoryProxy->getIncompleteId());
+      return Result(Result::UNKNOWN, UnknownExplanation::INCOMPLETE);
+    }
+  }
+  else if (d_theoryProxy->isUnsound())
+  {
+    outputIncompleteReason(UnknownExplanation::UNSOUND,
+                           d_theoryProxy->getUnsoundId());
+    return Result(Result::UNKNOWN, UnknownExplanation::UNSOUND);
   }
```

Comment:
I think it'd be better to merge these by having `TheoryProxy` have just a "isUnsound" method which would trigger the unknown response and a "getUnsoundId" or something we could say whether it's solution_unsound or refutation_unsound.

Reply:
TheoryEngine needs to track both things.  I think this is the right place to process the result, where TheoryEngine just implements simple getters for whether it is model/refutation sound.